src/Compilers/CSharp/Portable/Emitter/EditAndContinue/CSharpSymbolMatcher.cs

```diff
@@ -718,6 +728,9 @@ private bool AreNamedTypesEqual(NamedTypeSymbol type, NamedTypeSymbol other)
                 // TODO: Test with overloads (from PE base class?) that have modifiers.
                 Debug.Assert(!type.HasTypeArgumentsCustomModifiers);
                 Debug.Assert(!other.HasTypeArgumentsCustomModifiers);
+                Debug.Assert(!type.IsTupleType);
+                Debug.Assert(!other.IsTupleType);
```

Comment:
Hum. Thanks for the clarification. This means there must still be an un-handled scenario.
The `AreMethodsEqual` method goes int type *arguments*. Somehow I haven't managed to hit that path with a tuple yet.